components/server/src/installation-admin/installation-admin-controller.ts

```diff
@@ -17,7 +18,11 @@ export class InstallationAdminController {
         const app = express();
 
         app.get("/data", async (req: express.Request, res: express.Response) => {
+            const spanCtx =
+                opentracing.globalTracer().extract(opentracing.FORMAT_HTTP_HEADERS, req.headers) || undefined;
+            const span = opentracing.globalTracer().startSpan("telemetryDataEndpoint", { childOf: spanCtx });
             res.status(200).json(await this.telemetryDataProvider.getTelemetryData());
```

Comment:
The pattern here is to wrap the actual code in a `try { code } finally { span.finish(); }` to make sure we do not leak resources.